diff -urN freeamp.org/Makefile.header.in freeamp/Makefile.header.in
--- freeamp.org/Makefile.header.in	Sat Apr 21 12:16:23 2001
+++ freeamp/Makefile.header.in	Sat Apr 21 12:16:45 2001
@@ -124,5 +124,5 @@
 
 @SET_MAKE@
 
-PIMAKE = $(MAKE) -f Makefile-plugins -$(MAKEFLAGS)
+PIMAKE = $(MAKE) -f Makefile-plugins $(MAKEFLAGS)
 
diff -urN freeamp.org/configure.in freeamp/configure.in
--- freeamp.org/configure.in	Sat Apr 21 12:16:23 2001
+++ freeamp/configure.in	Sat Apr 21 12:18:08 2001
@@ -740,21 +740,6 @@
   AC_DEFINE(HAVE_ZLIB)
 )
 AC_SUBST(ID3_LIBS)
-
-AC_CHECK_HEADERS(kde/artsc/artsc.h, have_arts="true", have_arts="false")
-
-if test "$have_arts" = "false"; then
-  AC_CHECK_HEADERS(artsc/artsc.h, have_arts="true", 
-                    have_arts="false")
-fi
-
-ARTS_LIBS=""
-AC_CHECK_LIB(artsc, arts_init, 
-  OSDEPPLUGINS="$OSDEPPLUGINS plugins/arts.pmo";
-  OSINC="$OSINC -I$srcdir/io/arts/include"
-  ARTS_LIBS="-lartsc -laudiofile"
-  AC_MSG_RESULT([compiling arts pmo plugin])
-)
 
 AC_SUBST(ARTS_LIBS)
 AC_SUBST(EXTRALIBS)
