--- zinf-2.2.4/configure.ac.orig	2003-08-02 22:34:57.000000000 +0200
+++ zinf-2.2.4/configure.ac	2003-08-17 13:30:38.846093000 +0200
@@ -172,8 +172,8 @@
 dnl Checks fore core libraries.
 dnl
 
-AC_CHECK_LIB(socket, connect, [EXTRALIBS="$EXTRALIBS -lsocket"])
-AC_CHECK_LIB(nsl, gethostbyname, [EXTRALIBS="$EXTRALIBS -lnsl"])
+AC_CHECK_FUNC(socket, , [AC_CHECK_LIB(socket, connect, [EXTRALIBS="$EXTRALIBS -lsocket"])])
+AC_CHECK_FUNC(gethostbyname, , [AC_CHECK_LIB(nsl, gethostbyname, [EXTRALIBS="$EXTRALIBS -lnsl"])])
 
 dnl Threads
 if test "$host_os" = "beos"; then
