--- zinf-2.2.4/configure.ac.orig	2003-08-02 22:34:57.000000000 +0200
+++ zinf-2.2.4/configure.ac	2003-08-17 13:30:38.846093000 +0200
@@ -172,8 +172,8 @@
 dnl Checks fore core libraries.
 dnl
 
-AC_CHECK_LIB(socket, connect, [EXTRALIBS="$EXTRALIBS -lsocket"])
-AC_CHECK_LIB(nsl, gethostbyname, [EXTRALIBS="$EXTRALIBS -lnsl"])
+AC_CHECK_FUNC(socket, , [AC_CHECK_LIB(socket, connect, [EXTRALIBS="$EXTRALIBS -lsocket"])])
+AC_CHECK_FUNC(gethostbyname, , [AC_CHECK_LIB(nsl, gethostbyname, [EXTRALIBS="$EXTRALIBS -lnsl"])])
 
 dnl Threads
 if test "$host_os" = "beos"; then
--- zinf-2.2.5/m4/pthread.m4.orig	2003-09-16 19:35:17.000000000 +0200
+++ zinf-2.2.5/m4/pthread.m4	2004-02-08 11:42:30.442073416 +0100
@@ -126,12 +126,13 @@
 
 
 dnl Check for SunOS thread library style 	
+AC_CHECK_FUNC(sched_get_priority_max,,[
 AC_CHECK_LIB(rt, sched_get_priority_max, 
 [
  AC_DEFINE(HAVE_SCHED_RT,,[Defined if -lrt is needed for RT scheduling])
  LIBS="$LIBS -lrt"
 ]
-)
+)])
 
 PTHREAD_LIBS="$LIBS"
 
